src/js/reducers/inspectorReducer.ts

```diff
@@ -1,93 +1,67 @@
-/* eslint new-cap:0 */
-'use strict';
+import {ActionType, getType} from 'typesafe-actions';
+import * as inspectorActions from '../actions/inspectorActions';
+import * as markActions from '../actions/markActions';
+import {ExpandedLayers, Inspector, InspectorRecord} from '../store/factory/Inspector';
 
-var Immutable = require('immutable'),
-    immutils = require('../util/immutable-utils'),
-    getIn = immutils.getIn,
-    setIn = immutils.setIn,
-    ACTIONS = require('../actions/Names');
-
-function expandLayers(state, layerIds) {
-  return layerIds.reduce(function(newState, layerId) {
-    return setIn(newState, 'encodings.expandedLayers.' + layerId, true);
+function expandLayers(state: InspectorRecord, layerIds: number[]): InspectorRecord {
+  return layerIds.reduce(function(newState: InspectorRecord, layerId: number) {
+    return newState.setIn(['encodings', 'expandedLayers', layerId], true);
```

Comment:
Do we expect `layerId` to be a string or number? In the old code, `layerId` would get coerced to a string. Shouldn't matter so long as we're consistent.